src/Pardiso.jl

```diff
@@ -241,6 +241,12 @@ function fix_iparm!(ps::AbstractPardisoSolver, T::Symbol)
     end
 end
 
+function isstructurallysymmetric(A::SparseMatrixCSC{Tv,Ti})  where {Ti, Tv <: PardisoNumTypes}
+    I, J, _ = findnz(A)
```

Comment:
I wonder if this is the most efficient way of doing it. Maybe it is worth copy pasting https://github.com/JuliaSparse/SparseArrays.jl/blob/9a4656134eece3483216ce1244bd0b3a20aee6f1/src/sparsematrix.jl#L4081-L4154 and remove the nzval checks?

Reply:
It was just the simplest I could think of on the spot! 😅 

I like your idea and I'm sure it's much more efficient, but shouldn't that code live in SparseArrays.jl instead? What about adding a third `check2::Function` argument that defaults to `identity` so that the tests become `check2(T[i,j]) == check(T[j,i])`, and then we could simply add
```julia
isstructurallysymmetric(A::AbstractSparseMatrixCSC) = is_hermsym(A, Returns(true), Returns(true))
```
Or is that a terrible idea?

Reply:
Yeah, it's a good idea to add to SparseArray,s but we still need something for earlier julia versions.

Reply:
The `check` function could be changed to a two arg one with `check(a,b) = a == b` for issymetric, `check(a,b) = a == conj(b)` for ishermitian and `check(a,b) = true` for isstructuralsymmetric?

Reply:
> Yeah, it's a good idea to add to SparseArray,s but we still need something for earlier julia versions.

Is that because in earlier Julia versions SparseArrays used to be a stdlib and so this change wouldn't work then?

Reply:
It is still an stdlib, it is just developed ina  separate repo.

Reply:
So how should I proceed? Should I just PR your 2-arg `check` in SparseArrays.jl? Or copy-paste the SparseArrays code here? Or do something else?

Reply:
In my opinion it is best to do both.

Reply:
As per your suggestion, I've copy-pasted the SparseArrays.jl code and modified it to work with a 2-arg `check` function.